### ISSUE TITLE
Random NullPointerException after upgrade to 4.9.2

### ISSUE THREAD

**Issue by bjornbak** [etorreborre/specs2#819]
Hi,

We upgraded recently from 4.8.3 to 4.9.2 after the upgrade we experience random NullPointerExceptions in the specs2 code.

Here is an example where com.company.package.MySpec is a spec we try to execute:
```
[2020-03-31T09:20:25.033Z] [error] cannot create an instance for class com.company.package.ml.MySpec
[2020-03-31T09:20:25.033Z] [error]   caused by java.lang.NullPointerException
[2020-03-31T09:20:25.033Z] [error]   sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2020-03-31T09:20:25.033Z] [error]   sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[2020-03-31T09:20:25.033Z] [error]   sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2020-03-31T09:20:25.033Z] [error]   java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.reflect.Classes$$anonfun$createInstanceForConstructor$1.apply(Classes.scala:74)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.reflect.Classes$class.org$specs2$reflect$Classes$$newInstance(Classes.scala:96)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.reflect.Classes$class.createInstanceForConstructor(Classes.scala:74)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.reflect.Classes$class.org$specs2$reflect$Classes$$findInstance(Classes.scala:58)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.reflect.Classes$class.createInstanceFromClass(Classes.scala:38)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.reflect.Classes$.createInstanceFromClass(Classes.scala:122)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.reflect.Classes$$anonfun$createInstance$2.apply(Classes.scala:31)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.reflect.Classes$$anonfun$createInstance$2.apply(Classes.scala:30)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Arrs.go$1(Eff.scala:386)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.IntoPolyLower2$$anon$3$$anonfun$apply$8.apply(IntoPoly.scala:60)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.IntoPolyLower2$$anon$3$$anonfun$apply$8.apply(IntoPoly.scala:60)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Arrs.go$1(Eff.scala:383)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Interpret$$anonfun$go$1$5.apply(Interpret.scala:198)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Interpret$$anonfun$go$1$5.apply(Interpret.scala:198)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Arrs.go$1(Eff.scala:383)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.CollectedUnions$$anonfun$othersEff$1.apply(Unions.scala:97)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.CollectedUnions$$anonfun$othersEff$1.apply(Unions.scala:97)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Arrs.go$1(Eff.scala:386)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Arrs.apply(Eff.scala:348)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.CollectedUnions$$anonfun$continuation$1.apply(Unions.scala:84)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.CollectedUnions$$anonfun$continuation$1.apply(Unions.scala:84)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Arrs.go$1(Eff.scala:383)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[2020-03-31T09:20:25.033Z] [error]   org.specs2.control.eff.Interpret$$anon$1$$anonfun$onEffect$1.apply(Interpret.scala:53)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.Interpret$$anon$1$$anonfun$onEffect$1.apply(Interpret.scala:53)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.fp.EitherOps$.bimap$extension(EitherSyntax.scala:82)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.Interpret$$anon$1.onEffect(Interpret.scala:53)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.Interpret$$anon$1.onApplicativeEffect(Interpret.scala:61)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.Interpret$$anon$1.onApplicativeEffect(Interpret.scala:45)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.Interpret$class.go$1(Interpret.scala:200)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.Interpret$class.interpretLoop(Interpret.scala:207)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.Interpret$.interpretLoop(Interpret.scala:635)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.Interpret$class.interpret(Interpret.scala:71)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.Interpret$.interpret(Interpret.scala:635)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.Interpret$class.interpret1(Interpret.scala:78)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.Interpret$.interpret1(Interpret.scala:635)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.ErrorInterpretation$class.runError(ErrorEffect.scala:87)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.ErrorEffect$.runError(ErrorEffect.scala:187)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.eff.syntax.error$ErrorEffectOps.runError(error.scala:14)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.ExecuteActions$class.executeActionFuture(ExecuteActions.scala:36)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.control.ExecuteActions$.executeActionFuture(ExecuteActions.scala:93)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.runner.SbtTask.executeFuture(SbtRunner.scala:142)
[2020-03-31T09:20:25.034Z] [error]   org.specs2.runner.SbtTask.execute(SbtRunner.scala:162)
[2020-03-31T09:20:25.034Z] [error]   sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:304)
[2020-03-31T09:20:25.034Z] [error]   java.util.concurrent.FutureTask.run(FutureTask.java:266)
[2020-03-31T09:20:25.034Z] [error]   java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2020-03-31T09:20:25.034Z] [error]   java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2020-03-31T09:20:25.034Z] [error]   java.lang.Thread.run(Thread.java:748)
```

It happens for random specs.

**Comment by etorreborre** [comment]
Thanks for reporting this. Do you need ScalaJS 1.0? If not you can stay on `4.8.3` then until I fix this. If you could create a small reproducible spec in the meantime that would greatly help :-)

**Comment by bjornbak** [comment]
Reverting but creating a reproducing specs is difficult as it is very random with maybe a couple of occurrences per 100 builds.

**Comment by etorreborre** [comment]
This might be some classloader issue. We have this that in the past. Hard to diagnose :-(

**Comment by symbiont-eric-torreborre** [comment]
I'm closing this issue for lack of activity (clearly on my side...)

**Comment by etorreborre** [comment]
I'm closing this issue since it seems to be solved by #824
